### DISCUSSION TITLE
NIRCam Coronagraph Model

### DISCUSSION THREAD

**Discussion by benjaminpope** [LouisDesdoigts/dLux#74] [Ideas]
It is worth looking at whether we can forwards-model the NIRCam coronagraph - this will require great precision at the 1e-5 contrast level! I don't think the MIRI one is as important.

**Comment by Jordan-Dennis** [comment]
[This](https://github.com/spacetelescope/webbpsf/blob/develop/webbpsf/optics.py#L680) link is for the NIRCam coronagraph model in `WebbPSF`. Are we able to simply steal the arrays from this or do we need to implement this in `jax`. Having skimmed the code I tend to think it will need some sought of re-implementation.

**Comment by LouisDesdoigts** [comment]
Yep this is an easy one - It just requires developing code to put propagators between the layers read out of the webbpsf planes. This will ultimately form a part of the webbpsf -> ∂Lux pipeline I'm building on the JWST_dev branch. 

I'm happy to tackle this as I'm basically already working on it 👍

**Reply by Jordan-Dennis** [reply]
Too easy
